drip/drips.py

```diff
@@ -20,25 +25,69 @@
 
 import logging
 
+DEFAULT_DRIP_MESSAGE_CLASS = 'drip.drips.DripMessage'
 
-def configured_message_classes() -> dict:
-    """[summary]
 
-    :return: [description]
-    :rtype: dict
+def configured_message_classes() -> ChainMap:
+    """
+    Returns a ChainMap (basically a dict), between default settings and
+    settings.DRIP_MESSAGE_CLASSES which should be a dictionary of the form:
+
+        { message_class_name: class_path }
+
+    For example, if you define this at DRIP_MESSAGE_CLASSES:
+
+        {
+            'default': 'my.path.to.DripMessage',
+            'my_other_class': 'my.other.path.to.DripMessage',
+        }
+
+    The returning value of this method would be:
+
+        {
+            'default': 'my.path.to.Drip',
+            'my_other_class': 'my.other.path.to.DripMessage',
+        }
+
+    But, if you define this:
+
+        {
+            'my_other_class': 'my.other.path.to.DripMessage',
+        }
+
+    The returning value of this method would be:
+
+        {
+            'default': 'drip.drips.DripMessage',
+            'my_other_class': 'my.other.path.to.DripMessage',
+        }
+
+    A 'default' key is added when it's not present on DRIP_MESSAGE_CLASSES.
     """
-    conf_dict = getattr(settings, 'DRIP_MESSAGE_CLASSES', {})
-    if 'default' not in conf_dict:
-        conf_dict['default'] = 'drip.drips.DripMessage'
-    return conf_dict
+    default_config = {'default': DEFAULT_DRIP_MESSAGE_CLASS}
+    user_defined_config = getattr(settings, 'DRIP_MESSAGE_CLASSES', {})
+    return ChainMap(user_defined_config, default_config)
 
 
-def message_class_for(name: str):
-    path = configured_message_classes()[name]
-    mod_name, klass_name = path.rsplit('.', 1)
-    mod = import_module(mod_name)
-    klass = getattr(mod, klass_name)
-    return klass
+@lru_cache()
+def message_class_for(name: str) -> Any:
+    """
+    Given a class' path, returns a reference to it.
+    Raises MessageClassNotFound exception if name is not present in
+    the ChainMap produced by configured_message_classes.
+    """
+    try:
+        path = configured_message_classes()[name]
+    except KeyError:
+        logging.error(
+            'Name "{}" not found in configured message classes.'.format(name)
+        )
+        raise MessageClassNotFound() from None
+    else:
```

Comment:
You don't need `else` here. If the exception is not raised, this block will be executed. Please remove the else

Reply:
I believe it adds more readability to the code, in the sense that it's clear that the original developer wanted that code inside the `else` clause to be executed *only* when no errors occur. But it's just a personal preference. If the team prefers some approach to the other we can use that one.